ISPN-12728 Add Persistent Memory to Soft Index File Store

wburns opened the pull request:
https://issues.redhat.com/browse/ISPN-12728

This adds in support for persistent memory usage. The mashona library will only return a non null FileChannel if it was able to allocate it with persistent memory. In the case it is null we just fall back to the prior code as normal.

Unfortunately, there is no way to test this currently. However, Jonathan Halliday tested this before. We may need to discuss further with him how we can properly test this long term though.

ryanemerson commented on persistence/soft-index/pom.xml:
```diff
@@ -17,6 +17,11 @@
+      <dependency>
+         <groupId>io.mashona</groupId>
+         <artifactId>mashona-logwriting</artifactId>
+         <version>1.0.0.Beta1</version>
```
I think we still need the version to be defined in the `build-configuration/pom.xml` so that it can easily be substituted downstream. The dependency should probably be defined in the root pom's `dependencyManagment` section as well.

@pruivo Is this right?

pruivo replied:
right, except the version/property is not mandatory to be in `build-configuration/pom.xml` since there is no plan to be consumed outside this repo.

wburns replied:
Ooops, I forgot to move this over. I was definitely planning on doing so.

wburns commented:
Updated and fixed compilation error in tools. I defaulted the max file size to 1MB. It should be fine for the migration tool I would expect. And this is only used with pmem.

tristantarrant commented:
The dependency should be optional

wburns commented:
> The dependency should be optional

Alright, I will change this over.

wburns commented:
Updated to make it optional.

ryanemerson commented:
Thanks @wburns

pruivo commented:
@wburns javadoc is broken
```
[ERROR] Exit code: 1 - /mnt/data/pedro/work/infinispan/persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/pmem/PmemUtilWrapper.java:7: error: package io.mashona.logwriting does not exist
[ERROR] import io.mashona.logwriting.PmemUtil;
[ERROR]                             ^
[ERROR] 
[ERROR] Command line was: /mnt/data/pedro/opt/java-11-sun/bin/javadoc -J--add-opens -Jjdk.javadoc/jdk.javadoc.internal.tool=ALL-UNNAMED @options @packages
```
